Automattic/wp-calypso
# Trail Map: create mapping of plans and trail map features

Related to #89682

## Proposed Changes

* This PR adds the mapping from Plan -> Feature list. This is done by checking the experiment variant in `get2023PricingGridSignupWpcomFeatures` and then returning the feature list accordingly. The default return value should be unchanged.
* `get2023PricingGridSignupJetpackFeatures` should return an empty array for any trail map variant.
* Also made some minor changes in `features-list.tsx`, `feature-group-plan-map.tsx` and `packages/calypso-products/src/main.ts` to match the strings according to the design.
* Added storybook stories for the trail map variants.
* In 78da7bd0a74e97eaf5016b6f13ea0290bab80f4f, renamed `includeAllFe
atures` -> `includePreviousPlanFeatures`

**Control**
![localhost_52190_iframe html_args= id=featuresgrid--trail-map-control viewMode=story](https://github.com/Automattic/wp-calypso/assets/5436027/dc1ef052-c9fd-42fb-b0b8-94bd1c461d58)

**Treatment Structure**
![localhost_52190_iframe html_args= id=featuresgrid--trail-map-structure viewMode=story](https://github.com/Automattic/wp-calypso/assets/5436027/1158355c-8a50-44f6-a2ad-fd95a8206414)

**Treatment Copy**
![localhost_52190_iframe html_args= id=featuresgrid--trail-map-copy viewMode=story](https://github.com/Automattic/wp-calypso/assets/5436027/e8e02b83-442f-4017-976f-fecf12bff867)

**Treatment Copy and Structure**
![localhost_52190_iframe html_args= id=featuresgrid--trail-map-copy-and-structure viewMode=story](https://github.com/Automattic/wp-calypso/assets/5436027/a76eb1fd-5e87-4f64-9eae-43e9350dba03)


## Testing Instructions



* Run `yarn workspace @automattic/plans-grid-next storybook`.
* Check each of the trail map stories and confirm that it matches the design (Uzd0AYiDAV4eFq1V35T9Ef-fi-800_19072)
* **Please Note**: The sub feature list for the `25+ pre-installed ecommerce plugins ($1,500/year value)` feature is in a follow-up PR: https://github.com/Automattic/wp-calypso/pull/90200

## Pre-merge Checklist
